src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java

```diff
@@ -227,7 +227,7 @@ public AbstractAnomalyDetectorActionHandler(
     public void start() {
         String resultIndex = anomalyDetector.getResultIndex();
         // use default detector result index which is system index
-        if (resultIndex == null) {
+        if (resultIndex == null || this.isDryRun) {
```

Comment:
If using custom index, we won't validate result index security permission in validateCustomResultIndexAndExecute, right? Is that by design? If so, why?

Reply:
This is initially to avoid creating custom result index when user call validation API. This change also skip the permission check for an existing custom result index. I'm ok to this as we still have validation when user call create detector API. But like Kaituo said, it will be better if we can call `validateCustomResultIndexAndExecute` to validate permission for an existing custom result index.

Reply:
Yes I can call `validateCustomResultIndexAndExecute` if `resultIndex` is not null and `isDryRun` is true. What are your thoughts on wrapping these exception and returning them as ADValidationResponse or to simply return as exception/error message that will show as bottom of page notification. Might need to have a longer discussion offline for best way going forward with these exceptions in general.

Reply:
I like your idea of returning them as ADValidationResponse.  We can extract the original exception's error message and return ADValidationResponse instead.